src/Symfony/Component/Form/ChoiceList/ArrayChoiceList.php

```diff
@@ -233,13 +233,13 @@ private function castableToString(array $choices, array &$cache = array())
                 }
 
                 continue;
-            } elseif (!is_scalar($choice)) {
+            } elseif (!is_scalar($choice) && null !== $choice) {
```

Comment:
I confirm this would work.

But @Tobion is still right, when having both a placeholder and a choice with an explicit `null` or empty string `''` value, there is a conflict.

The default `null` choice is pre selected over the `placeholder` added in the choice list, so perhaps it should not be added in this case special case :
if default data is `null` is && `''` in choice values then `placeholder` would be set to `false`.
There would be no need then to reverse `elseif (!is_scalar($choice) && null !== $choice)` in a new commit.
To have a choice pre selected __and__ a placeholder looks odd to me, especially with select input.
Also the use case for a `null` value in a choice list make me guess it's about handling a default value in some way.
Does it makes sense to you ? ping __@symfony/deciders__

Otherwise, @Tobion suggested this mapping :
>     false -> '0'
    null -> '~'

This would solve it indeed, but may not result in an expected behaviour when using custom submission (e.g javascript) as one would send an empty value and want it to be mapped to the choice with a `null` value instead of needing to send `'~'`.

The only other way would I can see would be to set a special string value for the `placeholder` in order to prevent it from being submitted and to fallback on default data . Just for the example something like "~sf_placeholder-value" that we know will not conflict with other string values.